File: ios/xue_hua_gaode_map/Sources/xue_hua_gaode_map/AmapPrivacyState.swift
import Foundation

enum AmapPrivacyState {
    private(set) static var privacyAgreed = false
    
    static func setPrivacyAgreed(_ agreed: Bool) {
        privacyAgreed = agreed
    }
    
    static func privacyError() -> NSError {
        NSError(
            domain: "GaodePrivacy",
            code: 1,
            userInfo: [
                NSLocalizedDescriptionKey:
                    "Gaode privacy compliance must be configured before using location or geofence APIs. " +
                "Call GaodeSdk.updatePrivacyShow and GaodeSdk.updatePrivacyAgree first.",
            ]
        )
    }
}

File: android/src/main/kotlin/com/kurban/xue_hua_gaode_map/AmapPrivacyState.kt
package com.kurban.xue_hua_gaode_map

object AmapPrivacyState {
    @Volatile
    var privacyAgreed: Boolean = false

    fun requirePrivacyAgreed() {
        if (!privacyAgreed) {
            throw IllegalStateException(
                "Gaode privacy compliance must be configured before using location or geofence APIs. " +
                        "Call GaodeSdk.updatePrivacyShow and GaodeSdk.updatePrivacyAgree first.",
            )
        }
    }
}
